projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kerberos_configuration.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import os
+
+from vdk.internal.core.config import Configuration
+from vdk.internal.core.config import ConfigurationBuilder
+
+KRB_AUTH = "KRB_AUTH"
+KEYTAB_FOLDER = "KEYTAB_FOLDER"
+KEYTAB_FILENAME = "KEYTAB_FILENAME"
+KEYTAB_PRINCIPAL = "KEYTAB_PRINCIPAL"
+KEYTAB_REALM = "KEYTAB_REALM"
+KERBEROS_KDC_HOST = "KERBEROS_KDC_HOST"
+
+
+class KerberosPluginConfiguration:
+    def __init__(self, job_name: str, job_directory: str, config: Configuration):
+        self.__job_name = job_name
+        self.__job_directory = job_directory
+        self.__config = config
+
+    def authentication_type(self):
+        return self.__config.get_value(KRB_AUTH)
+
+    def keytab_folder(self):
+        keytab_folder = self.__config.get_value(KEYTAB_FOLDER)
+        if keytab_folder is None:
+            keytab_folder = self.__job_directory
+        return keytab_folder
+
+    def keytab_filename(self):
+        keytab_filename = self.__config.get_value(KEYTAB_FILENAME)
+        if keytab_filename is None:
+            keytab_filename = f"{self.__job_name}.keytab"
+        return keytab_filename
+
+    def keytab_pathname(self):
+        return os.path.join(self.keytab_folder(), self.keytab_filename())
+
+    def keytab_principal(self):
+        keytab_principal = self.__config.get_value(KEYTAB_PRINCIPAL)
+        if keytab_principal is None:
+            keytab_principal = f"pa__view_{self.__job_name}"
+        return keytab_principal
+
+    def keytab_realm(self):
+        return self.__config.get_value(KEYTAB_REALM)
+
+    def kerberos_host(self):
+        return self.__config.get_value(KERBEROS_KDC_HOST)
+
+
+def add_definitions(config_builder: ConfigurationBuilder) -> None:
+    config_builder.add(
+        key=KRB_AUTH,
+        default_value=None,
+        description="Specifies the Kerberos authentication type to use. "
+        "Possible values are 'minikerberos' and 'kinit'. "
```

Comment:
When I checkout locally and reformat code, files look a bit different. Cannot remember, do we have any auto reformatting applied on commit?

Reply:
We do have pre-commit hooks that take care of formatting.